unify gradient methods; (#71)
jump-dev/DiffOpt.jl: README.md

obtain product of jacobians with respect to problem parameters and a backward
pass vector.

> Currently, DiffOpt supports two backends. Firstly, if the optimization problem is of quadratic form i.e.

Reviewer: ```suggestion
Currently, DiffOpt supports two backends. If the optimization problem is of quadratic form i.e.
```
Firstly not needed here

```
minimize_z z^T Q z / 2 + q^T z
subject to: Az = b,
            Gz ≤ h
```
then one can compute gradients by providing a backward pass vector
```julia

> bpv = [1.0 1.0]

Reviewer: ```suggestion
bpv = [1.0, 1.0]
```

grads = backward(diff, ["Q", "q", "h"], bpv)
```

Secondly, for a conic problem of the format:
```
minimize_x c^T x

> subject to: Ax + b \in K

Reviewer: ```suggestion
subject to: Ax + b in K
```

```
where

> - `the obejctive is linear

Reviewer: ```suggestion
- `the objective is linear
```

> - `K` is a cone composed of linear, Semidefinite, Second Order constraints

Reviewer: ```suggestion
- `K` is a Cartesian product of linear, semidefinite, second-order cones
```
